x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/BaseElasticsearchInternalService.java

```diff
@@ -66,11 +74,12 @@ public BaseElasticsearchInternalService(InferenceServiceExtension.InferenceServi
     // service package.
     public BaseElasticsearchInternalService(
         InferenceServiceExtension.InferenceServiceFactoryContext context,
-        Consumer<ActionListener<Set<String>>> platformArchFn
+        Consumer<ActionListener<PreferredModelVariant>> platformArchFn
```

Comment:
The name `platformArchFn` seems outdated now that this consumes a `PreferredModelVariant`